modules/developer_manual/pages/core/apis/roles-api.adoc

```diff
@@ -0,0 +1,95 @@
+= Roles API
+:toc: right
+:toclevels: 1
+:request-base-path: ocs/v1.php/cloud/roles
+:accept-language-header-url: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language
+
+== Introduction
+
+The roles API allows clients to ask the server for supported roles.
+Three things are worth noting about this API:
+
+. Only roles for public links are implemented.
+. Apps can listen to the event dispatcher and add their own roles.
+. Apps cannot change existing roles.
+
+== Get Roles
+
+[cols=",,",options="header",]
+|=================================================================
+| Request Path 
+| Method 
+| Content Type
+
+| `ocs/v1.php/cloud/roles` 
+| `GET` 
+| `text/xml` or `text/json`
+|=================================================================
+
+=== Request Parameters
+
+[cols="15%,15%,70%",options="header",width=100%]
+|===
+|Attribute 
+|Type 
+|Description
+
+|`format` 
+|string 
+|The format of the response content. 
+Accepted values are `xml` and `json`.
+|===
+
+=== Setting The Language of the Response Body
+
+The language of the response’s content can be set with {accept-language-header-url}[the Accept-Language header].
+By default, the response will be in English. 
+You can see an example of requesting the response in a specific language in the code example below.
+
+=== Code Example
+
+[tabs]
+====
+Curl::
++
+--
+[source,console,subs="attributes+"]
+----
+#!/usr/bin/env bash
+
+API_PATH="ocs/v1.php/cloud/roles?format=json"
+SERVER_URI="{oc-examples-server-url}"
+
+curl '$SERVER_URI/$API_PATH/' \
+  --user "{oc-examples-username}:{oc-examples-password}" \
+  -H 'Accept-Language: de-DE' | jq
```

Comment:
having de-DE here and showing english texts in the response is missleading.

Reply:
Fair call. I'll update that.